lib/nodejs/esm-utils.js

```diff
@@ -39,48 +39,34 @@ exports.requireOrImport = async (file, esmDecorator) => {
     return formattedImport(file, esmDecorator);
   }
   try {
-    return dealWithExports(await formattedImport(file, esmDecorator));
-  } catch (err) {
+    return require(file);
+  } catch (requireErr) {
     if (
-      err.code === 'ERR_MODULE_NOT_FOUND' ||
-      err.code === 'ERR_UNKNOWN_FILE_EXTENSION' ||
-      err.code === 'ERR_UNSUPPORTED_DIR_IMPORT'
+      requireErr.code === 'ERR_REQUIRE_ESM' ||  requireErr.code === 'ERR_INTERNAL_ASSERTION' ||
+      (requireErr instanceof SyntaxError &&
+        requireErr
+          .toString()
+          .includes('Cannot use import statement outside a module'))
     ) {
+      // In Node.js environments after version 22.11, the `loadESMFromCJS` function 
+      // is used within the `require` function to handle cases where the target file 
+      // is in ESM (ECMAScript Module) format. If the Node.js version is after 22.11, 
+      // the code will import the module without any issues. For older versions, 
+      // this `if` statement is required to properly handle ESM modules.
+      // This `if` statement can be removed once all Node.js environments with current 
+      // support include the `loadESMFromCJS` function.
```

Comment:
```suggestion
      // This `if` statement can be removed once Mocha only supports Node.js versions
      // that allow require(ESM): Node.js ^20.19.0 || >=22.12.0
```

That `20.19.0` is a guess, the wonderful https://github.com/nodejs/node/pull/56927 hasn't been released yet.